### PR TITLE
Fix formatting of arguments when break-fun-decl=fit-or-vertical

### LINKED ISSUE #877: function arguments formatted incorrectly
Input:
```
let fffffffffffffffffffffffffffffffffff x
      yyyyyyyyyyyyyyyyyyyyyyyyyyy =
  ()
```

Output:
```
let fffffffffffffffffffffffffffffffffff x
                                        yyyyyyyyyyyyyyyyyyyyyyyyyyy = ()
```

This function shouldn't need to be line-wrapped at all, it fits on one line.

This also isn't how I expect line-wrapping for function arguments to work when it does happen, but that may be down to whatever bug is causing the wrapping, or due to different configuration I'm used to. First I just want to stop this from wrapping.

Command-line:
`ocamlformat --profile=janestreet --enable-outside-detected-project unformatted.ml > formatted.ml`

I've seen this as early as 722240710c8ac5066862188b27a90a709ebb325c and as recently as 9f0bde1a8138eedaff4d844605d048f712703e90; I haven't checked other commits.

### PR DESCRIPTION
**Pull request by gpetiot**
Fix #877, maybe it could be improved to not have `:` alone on a line, but as a whole the formatting is more consistent.
And not using `Location.is_single_line` is a plus.

@Julow since you introduced this option, feel free to directly commit to this PR.

### CODE REVIEW

File: src/Fmt_ast.ml [hunk 1]
```diff
@@ -140,9 +140,7 @@ let fmt_expressions c width sub_exp exprs fmt fmt_expr =
 let wrap_fun_decl_args ~stmt_loc c k =
```
Comment (Julow) [inline src/Fmt_ast.ml]: `~stmt_loc` is not used.

File: src/Fmt_ast.ml [hunk 2]
```diff
@@ -2439,11 +2437,15 @@ and fmt_class_expr c ?eol ?(box = true) ({ast= exp; _} as xexp) =
   | Pcl_fun _ ->
       let xargs, xbody = Sugar.cl_fun c.cmts xexp in
       let stmt_loc = Sugar.args_location xargs in
+      let args_box =
```
Comment (Julow) [inline src/Fmt_ast.ml]: This is duplicated every time. I suggest returning the `args_box` from `wrap_fun_decl_args`.

File: src/Fmt_ast.ml (hunk 1 above)
Comment (gpetiot) [reply src/Fmt_ast.ml]: Fixed.

File: src/Fmt_ast.ml (hunk 2 above)
Comment (gpetiot) [reply src/Fmt_ast.ml]: Done.

**Comment by gpetiot** [comment]
The diff has now been reduced on tests, I'm running some tests with test_branch but it looks good so far.

**Comment by gpetiot** [comment]
Yes it will be done in another PR.

**Pull request merged**